src/API/AbstractAPIClient.php

```diff
@@ -39,42 +39,74 @@ public function callAPI(Request $request)
-            $bodyType = (($contentTypeHeader === self::APPLICATION_JSON_KEY) ? 'json' : 'body');
-
-            $requestOptions = array(
-                'headers' => $request->getHeaders(),
-                'query' => $request->getParameters(),
-                $bodyType => $request->getBody(),
-            );
+            $body = $request->getBody();
 
-            if ($this->config->getValue('debug')) {
-                $requestOptions['debug'] = fopen('php://stderr', 'w');
+            $isPaginatable = false;
+            if ($method === 'GET') {
+                $isPaginatable = true;
             }
 
-            $apiRequest = $client->createRequest($request->getMethod(), $request->getUrl(), $request->getHeaders(), $request->getBody(), $request->getParameters());
+            $mergedResponse = null;
 
-            $method = $request->getMethod();
+            $currentPage = 1;
+            $totalPages = 1;
 
-            // Since Guzzle automatically overwrites a new header when the request
-            // is POST / PATCH / DELETE / PUT, we need to overwrite the Content-Type header
-            // with setBody() function.
-            if ($method !== 'GET') {
-                $apiRequest->setBody(json_encode($request->getBody()), 'application/json');
-            }
+            while ($totalPages >= $currentPage) {
+                $apiRequest = $client->createRequest($method, $url, $headers, $body, array());
 
-            $response = $client->send($apiRequest)->json();
+                // Enable pagination
+                if ($isPaginatable) {
+                    $params['page'] = $currentPage;
+                }
 
-            if (json_last_error() !== JSON_ERROR_NONE) {
-                throw new BadResponseException('Error decoding client API JSON', $response);
-            }
+                // Assign parameters
+                $query = $apiRequest->getQuery();
+                foreach ($params as $key => $value) {
+                    $query->set($key, $value);
+                }
+
+                // Since Guzzle automatically overwrites a new header when the request
+                // is POST / PATCH / DELETE / PUT, we need to overwrite the Content-Type header
+                // with setBody() function.
+                if ($method !== 'GET') {
+                    $apiRequest->setBody(json_encode($request->getBody()), 'application/json');
+                }
+
+                $response = $client->send($apiRequest)->json();
+
+                if (json_last_error() !== JSON_ERROR_NONE) {
+                    throw new BadResponseException('Error decoding client API JSON', $response);
+                }
+
+                if (!$this->responseOk($response)) {
+                    $this->logAPICall($this->getAPIClientName(), array('type' => 'response', 'body' => $response), true);
+                }
+
+                if ($isPaginatable && isset($response['result_info'])) {
```

Comment:
These booleans are redundant - you can just check `isset($response['result_info'])`. Its impossible to have a pageable request in the V4 API without including this key.